### PR TITLE
Fix violations of xUnit analyzers

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Turning on a few warnings and fixing violations of them.

### CODE REVIEW

File: src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs [hunk 1]
```diff
@@ -116,23 +116,11 @@ public async Task TestNullFilePaths()
             Assert.Null(await storage.ReadStreamAsync(document, streamName));
         }
 
-        [Fact]
-        public async Task PersistentService_Solution_WriteReadDifferentInstances()
+        [Theory]
```
Comment (jasonmalinowski) [inline src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs]: @heejaechang This change of the pattern does have one implicit behavior change: before the same Solution object was being used for each "underlying test", and now it's creating a new one each time. I suspect that's fine but want your confirmation.

File: src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs [hunk 2]
```diff
@@ -332,71 +272,35 @@ private void DoSimultaneousWrites(Func<string, Task> write)
...
 
             using var storage = GetStorage(solution);
             Assert.True(await storage.WriteStreamAsync(streamName1, EncodeString(GetData1(size)), GetChecksum1(withChecksum)));
             DoSimultaneousReads(async () => ReadStringToEnd(await storage.ReadStreamAsync(streamName1, GetChecksum1(withChecksum))), GetData1(size));
         }
 
-        [Fact]
-        public async Task PersistentService_Project_SimultaneousReads()
+        [Theory]
+        [CombinatorialData]
```
Comment (jaredpar) [inline src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs]: >CombinatorialData [](start = 9, length = 17)

What is this? I can't find a good reference for it.

Comment (sharwell) [reply src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs]: It's a helper from @AArnott. It generates combinations of inputs (e.g. a `bool` parameter would run with both `true` and `false`, and an enum parameter would run with each enum value).

https://github.com/AArnott/Xunit.Combinatorial

File: src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs (hunk 1 above)
Comment (heejaechang) [reply src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs]: ya, it should be fine. it care value to be same not reference.

**Pull request merged**